### DISCUSSION TITLE
API: `{"non_field_errors":["Expected a list of items but got type \"dict\"."]}`

### DISCUSSION THREAD

**Discussion by tacerus** [netbox-community/netbox#8220] [Q&A] [answered]
Hi,
I am trying to `PATCH` a virtual machine object. Would someone be so kind to help me as to what is wrong with my syntax? I tried to follow the example in https://netbox.readthedocs.io/en/stable/rest-api/overview/, but seem to be missing something.

```
$ curl --data '{"vcpus": "1"}' -H "Content-Type: application/json" -H @token.txt -X PATCH 'https://netbox.example.com/api/virtualization/virtual-machines/?name=playground01'
{"non_field_errors":["Expected a list of items but got type \"dict\"."]}⏎
```

`GET` requests work flawlessly, so I assume my JSON is wrong or is wrongly submitted.

Thanks a lot for any hints.

**Comment by candlerb** [comment, marked as answer]
`https://netbox.example.com/api/virtualization/virtual-machines/` is the resource for "all virtual machines"

`https://netbox.example.com/api/virtualization/virtual-machines/?name=playground01` is the resource for "all virtual machines", filtered to a subset of zero or more.  It doesn't matter if it happens to be exactly one.

In order to PATCH a VM you need to use the resource for an individual VM, identified by its database ID:
`https://netbox.example.com/api/virtualization/virtual-machines/NNN/`

**Reply by tacerus** [reply]
Thank you very much!
